src/test/kotlin/org/opensearch/replication/repository/RestoreContextTests.kt

```diff
@@ -18,32 +21,38 @@ class RestoreContextTests : OpenSearchTestCase() {
         // given
         val fileName = "test_file"
 
-        val mockShard = mock(IndexShard::class.java)
-        val mockIndexCommit = mock(GatedCloseable::class.java) as GatedCloseable<IndexCommit>
-        val mockStore = mock(Store::class.java)
-        val mockDirectory = mock(org.apache.lucene.store.Directory::class.java)
-        val mockBaseInput = mock(IndexInput::class.java)
-        val mockClonedInput = mock(IndexInput::class.java)
+        val mockShard = mock<IndexShard>()
+        val mockIndexCommit = mock<GatedCloseable<IndexCommit>>()
+        val mockDirectory = mock<Directory>()
+        val mockBaseInput = mock<IndexInput>()
+        val mockClonedInput = mock<IndexInput>()
+        val mockStore = mock<Store> {
```

Comment:
The mocking process using `mockitokotlin2` doesn’t seem to be smooth.
Some mock objects end up being null, so I tried using `org.mockito.kotlin`, which is known to work better with Kotlin.

I think it has already been merged into https://github.com/mockito/mockito-kotlin.